newdle/api.py

```diff
@@ -77,6 +81,38 @@ def require_token():
     g.user = user
 
 
+@api.route('/avatar/<email>')
+@allow_anonymous
+def user_avatar(email):
+    size = request.args.get('size')
+    email_hex = hashlib.md5(email.encode()).hexdigest()
+    # make gravat return 404 HTTP code instead of a default image
+    query_args = {'d': '404'}
+    if size is not None:
+        query_args['s'] = size
+    gravatar_url = (
+        f'https://gravatar.com/avatar/{quote(email_hex)}?' f'{urlencode(query_args)}'
+    )
+    resp = requests.get(gravatar_url)
+    if resp.status_code == 404:
+        user = search_users(None, email, 1)[1][0]
```

Comment:
pass the initial via URL - doing a search here kills performance

Reply:
I was thinking about it as well. Would it be ok to something like `/api/avatar/<email>/<initial>`?